tmail-backend/webadmin/webadmin-rate-limit/src/main/java/com/linagora/tmail/webadmin/RateLimitPlanUserRoutes.java

```diff
@@ -0,0 +1,192 @@
+    private static final String REVOKE_PLAN_OF_USER_PATH = USERS_PATH + SEPARATOR + USERNAME_PARAM + RATE_LIMIT_BASE_PATH;
+    private static final String GET_PLAN_OF_USER_PATH = USERS_PATH + SEPARATOR + USERNAME_PARAM + RATE_LIMIT_BASE_PATH;
+
+    private final RateLimitingPlanUserRepository planUserRepository;
+    private final RateLimitationPlanRepository planRepository;
+    private final UsersRepository usersRepository;
+    private final JsonTransformer jsonTransformer;
+
+    @Inject
+    public RateLimitPlanUserRoutes(RateLimitingPlanUserRepository planUserRepository, RateLimitationPlanRepository planRepository,
+                                   UsersRepository usersRepository, JsonTransformer jsonTransformer) {
+        this.planUserRepository = planUserRepository;
+        this.planRepository = planRepository;
+        this.usersRepository = usersRepository;
+        this.jsonTransformer = jsonTransformer;
+    }
+
+    @Override
+    public String getBasePath() {
+        return RATE_LIMIT_BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(ATTACH_PLAN_TO_USER_PATH, attachPlanToUser(), jsonTransformer);
+        service.get(GET_USERS_OF_PLAN_PATH, getUsersOfPlan(), jsonTransformer);
+        service.get(GET_PLAN_OF_USER_PATH, getPlanOfUser(), jsonTransformer);
+        service.delete(REVOKE_PLAN_OF_USER_PATH, revokePlanOfUser(), jsonTransformer);
+    }
+
+    private Route attachPlanToUser() {
+        return (request, response) -> {
+            Username username = extractUsername(request);
+            RateLimitingPlanId planId = extractPlanId(request);
+            userPreconditions(username);
+            planIdPreconditions(planId);
+            return Mono.from(planUserRepository.applyPlan(username, planId))
+                .then(Mono.just(Responses.returnNoContent(response)))
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Route getUsersOfPlan() {
+        return (request, response) -> {
+            RateLimitingPlanId planId = extractPlanId(request);
+            planIdPreconditions(planId);
+            return Flux.from(planUserRepository.listUsers(planId))
+                .map(Username::asString)
+                .collectList()
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
```

Comment:
NPE spotted.

In good code NPEs are not supposed to happen cause we do not use null!